react-toggle-switch-component/src/toggleSwitch.jsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+class ToggleSwitch extends React.Component {
+    constructor(props){
+        super(props)
+        this.state = {clicked: false, classColor: 'default', onOff: 'On'}
+        this.toggle = this.toggle.bind(this);
+    }
+
+    toggle(){
+        if(this.state.clicked === true){
+            this.state.clicked = false;
```

Comment:
Don't change state properties directly, add this in with your `setState` on the next line.

```JSX
this.setState({
  onOff:'Off', classColor: 'black', clicked:false
});
```